### ISSUE TITLE
NullReferenceException in HttpTest.RespondWithJson in multithreaded environment

### ISSUE THREAD

**Issue by NathanTurnbow** [tmenier/Flurl#371]
See https://github.com/jordansjones/Draft/issues/17 for a concrete example

**Comment by tmenier** [comment]
Sorry for the delayed response. You are correct that `FlurlHttp.GlobalSettings` is not thead-safe. As a global static thing that is basically just its nature and not a bug, although what you ran into could certainly use some null checks.

I looked at your code, and I think what you should be doing [here](https://github.com/jordansjones/Draft/blob/7e5c415fa144c65776197e643abcd59c3e4ab65c/tests/Draft.Tests/Tests/VerificationStrategies/BaseVerificationStrategyTests.cs#L52-L62) is using `HttpTest.Settings` rather than global settings. Test-level [settings](https://flurl.io/docs/configuration/) always override those set at the global, client, or request level, within the logical call context of that `HttpTest`'s lifetime.

I think the best fix is to change your `InitializeInvalidHostHelper` implementation to something like this:

```c#
protected HttpTest InitializeInvalidHostHelper(Func<HttpTest, HttpRequestMessage, HttpResponseMessage> responseFactory = null)
{
    return new HttpTest().Configure(settings =>
        settings.HttpClientFactory = new TestingHttpClientFactory(responseFactory));
}
```

This way those settings will only apply within the context of the test that's using the returned `HttpTest`, even if you have other tests running in parallel. And once that `HttpTest` is disposed or goes out of scope, so do those settings, so I don't think you'd even need `ResetInvalidHostHelper` anymore.
